src/HotChocolate/Core/src/Abstractions/Execution/Path/IPathFactory.cs

```diff
@@ -0,0 +1,23 @@
+namespace HotChocolate.Execution;
+
+/// <summary>
+/// A factory for the different kinds of <see cref="Path"/>
+/// </summary>
+public interface IPathFactory
```

Comment:
do we need the interface or should we just have the base class and call the base class PathFactory?